test/MetaService.spec.ts

```diff
@@ -1,47 +1,134 @@
+      const actual = meta.hasMemory();
+      expect(actual).toEqual(false);
+    });
+    it('should return false when memory is not an Object', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      meta.memory = [];
+      const actual = meta.hasMemory();
+      expect(actual).toEqual(false);
+    });
+    it('should return false when memory is an empty Object', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      meta.memory = {};
+      const actual = meta.hasMemory();
+      expect(actual).toEqual(false);
+    });
+    it('should return true when memory is a non-empty Object', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      meta.memory = {test: 'test'};
+      const actual = meta.hasMemory();
+      expect(actual).toEqual(true);
+    });
+  });
+
+  describe('function: getSubFields', () => {
+    it('should be defined', () => {
+      const meta: MetaService = new MetaService('EarnCodeGroup');
+      const actual = meta.getSubFields;
+      expect(actual).toBeDefined();
+    });
+    it('should return array of fields case 1', () => {
+      const meta: MetaService = new MetaService('EarnCodeGroup');
+      const field = 'jobOrders(id,title,status)';
+      const res = meta.getSubFields(field);
+      expect(res[0]).toBe('id');
+      expect(res[1]).toBe('title');
+      expect(res[2]).toBe('status');
+    });
+    it('should return array of fields case 1', () => {
+      const meta: MetaService = new MetaService('EarnCodeGroup');
+      const field = 'jobOrders(id,title,status)';
+      const res = meta.getSubFields(field);
+      expect(res[0]).toBe('id');
+      expect(res[1]).toBe('title');
+      expect(res[2]).toBe('status');
+    });
+    it('should return array of fields case 2', () => {
+      const meta: MetaService = new MetaService('EarnCodeGroup');
+      const field = 'jobOrders[3](id,title,status)';
+      const res = meta.getSubFields(field);
+      expect(res[0]).toBe('id');
+      expect(res[1]).toBe('title');
+      expect(res[2]).toBe('status');
+    });
+    it('should return array of fields case 3', () => {
+      const meta: MetaService = new MetaService('EarnCodeGroup');
+      const field = 'jobOrders{status=\'closed\'}(id,title,status)';
+      const res = meta.getSubFields(field);
+      expect(res[0]).toBe('id');
+      expect(res[1]).toBe('title');
+      expect(res[2]).toBe('status');
+    });
+    it('should return array of fields case 3', () => {
+      const meta: MetaService = new MetaService('EarnCodeGroup');
+      const field = 'jobOrders(id, title, status)';
+      const res = meta.getSubFields(field);
+      expect(res[0]).toBe('id');
+      expect(res[1]).toBe('title');
+      expect(res[2]).toBe('status');
+    });
+    it('should return array of fields case 4', () => {
+      const meta: MetaService = new MetaService('EarnCodeGroup');
+      const fields = ' businessSectors[3](name,id){name=\'Insurance\'}, category';
```

Comment:
Are you testing this use case with at space and "category" specifically? seems unintentional.